File: Daily Resources/Day05/DSA - Prefix and Difference Arrays Solutions/range_xor_queries.py
n, q = map(int, input().split())
arr = list(map(int, input().split()))

# Compute prefix XORs (1-based)
prefix = [0] * (n + 1)
for i in range(n):
    prefix[i + 1] = prefix[i] ^ arr[i]

# Process queries
for _ in range(q):
    a, b = map(int, input().split())
    print(prefix[b] ^ prefix[a - 1])

File: Daily Resources/Day05/DSA - Prefix and Difference Arrays Solutions/range_xor_queries.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, q;
    cin >> n >> q;
    
    vector<int> arr(n);
    vector<int> prefix(n + 1, 0);  // prefix[0] = 0

    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
        prefix[i + 1] = prefix[i] ^ arr[i];
    }

    while (q--) {
        int a, b;
        cin >> a >> b;
        cout << (prefix[b] ^ prefix[a - 1]) << '\n';
    }

    return 0;
}
